lib/faye/redis.rb

```diff
@@ -205,11 +210,16 @@ def redis
           @server.info "Faye::Redis: redis connection reconnected"
         end
         connection.on(:reconnect_failed) do |count|
-          @server.info "Faye::Redis: redis connection reconnect failed (#{count}/4)"
+          @server.info "Faye::Redis: redis connection reconnect failed #{count} time(s)"
+          begin
+            fn = @options[:reconnect_failed] || ->(_, _) {}
+            fn.call(count, 'redis')
+          rescue => e
+            @server.error "Faye::Redis: Execution of reconnect_failed lambda failed with #{e} (redis connection)"
```

Comment:
The begin/call lambda/rescue behavior is duplicated in both `reconnect_failed` events.
You might want to extract it to a private method (something like `safely_call_reconnect_failed`), up to you.

Reply:
Fair enough. I'll remove duplication in the way we handle vents on connection all together and rerun the tests.